Cesium3DTiles/src/RasterizedPolygonsOverlay.cpp

```diff
@@ -0,0 +1,222 @@
+
+#include "Cesium3DTiles/RasterizedPolygonsOverlay.h"
+#include "Cesium3DTiles/BoundingVolume.h"
+#include "Cesium3DTiles/RasterOverlayTileProvider.h"
+#include "Cesium3DTiles/spdlog-cesium.h"
+#include "CesiumAsync/AsyncSystem.h"
+#include "CesiumAsync/IAssetAccessor.h"
+#include "CesiumGeospatial/GlobeRectangle.h"
+#include "CesiumUtility/IntrusivePointer.h"
+#include <memory>
+#include <string>
+
+namespace Cesium3DTiles {
+namespace {
+void rasterizePolygons(
+    CesiumGltf::ImageCesium& image,
+    const CesiumGeospatial::GlobeRectangle& rectangle,
+    const std::string& textureTargetName,
+    const std::vector<CartographicSelection>& cartographicSelections) {
+
+  double rectangleWidth = rectangle.computeWidth();
+  double rectangleHeight = rectangle.computeHeight();
+
+  // create source image
+  image.width = 256;
+  image.height = 256;
+  image.channels = 1;
+  image.bytesPerChannel = 1;
+  image.pixelData.resize(65536);
+
+  std::memset(image.pixelData.data(), 0, 65536);
+
+  // TODO: this is naive approach, use line-triangle
+  // intersections to rasterize one row at a time
+  // NOTE: also completely ignores antimeridian (really these
+  // calculations should be normalized to the first vertex)
+  for (const CartographicSelection& selection : cartographicSelections) {
+
+    if (selection.getTargetTextureName() != textureTargetName) {
+      continue;
+    }
+
+    const std::vector<glm::dvec2>& vertices = selection.getVertices();
+    const std::vector<uint32_t>& indices = selection.getIndices();
+    for (size_t triangle = 0; triangle < indices.size() / 3; ++triangle) {
```

Comment:
this looks like it tries to rasterize all the triangles in the CartographicSelection; however, there is a good chance that majority of the triangles are outside the bound of the tile and needs not to be rasterized.

Reply:
Another thing is if you can determine that the tile is completely bound by a triangle, it can return early with white image instead of doing any rasterization